### PR TITLE
Add a pull request template with important information for contributors.

### PR DESCRIPTION
**Pull request by pchote**
The goal here is to help contributors help themselves.

See https://github.com/blog/2111-issue-and-pull-request-templates and #13774.

### CODE REVIEW

**Comment by penev92** [comment]
I feel like we need to include notes about extra commits, commit messages and something along the lines of "if you need help/are unsure as to how to proceed, pop into IRC". Otherwise I like the idea.

**Comment by pchote** [comment]
I agree, especially about how commits should be structured.

**Comment by pchote** [comment]
Updated.

**Review by abcdefg30 — COMMENTED** [review]
Looks good to me already. I'd really like it though if we managed to include a link to http://webchat.freenode.net/?channels=openra somehow.

**Comment by penev92** [comment]
"If you need any help you can ask for help"  hmmm...

**Comment by reaperrr** [comment]
> "If you need any help you can ask for help" hmmm...

Just ditching the "for help" should do, in my opinion.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Mailaender) [inline .github/PULL_REQUEST_TEMPLATE.md]: That doesn't sound very welcoming. Why don't we just promote more long-term contributors to maintainer status?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Mailaender) [inline .github/PULL_REQUEST_TEMPLATE.md]: The tooling should do this automatically. This should never be a manual process.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (Mailaender) [inline .github/PULL_REQUEST_TEMPLATE.md]: Hard to read and misses punctuation:

```diff
- If you are changing shared mod or engine code make sure that you have tested your changes in all four default mods.
+ If you are changing shared mod or engine code, make sure that you have tested your changes in all four default mods.
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (pchote) [reply .github/PULL_REQUEST_TEMPLATE.md]: This statement is the reality that we have to face.  While we do have enthusiastic newer reviewers like @ltem, @MustaphaTR and @Smittytron helping out where they can, there are many PRs that they do not feel qualified to review.  Most non-trivial PRs sit for multiple weeks waiting for review, and stating this up front shows that this is not a personal snub against the contributor.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (pchote) [reply .github/PULL_REQUEST_TEMPLATE.md]: Once again, this statement reflects the reality of the situation.  Many PRs from new contributors sit with failing tests until a reviewer points them out, and there are other cases (such as tabs vs spaces or project file changes) that are not tested by the automatic tooling.

Comment (Mailaender) [reply .github/PULL_REQUEST_TEMPLATE.md]: In that case I suggest promoting @EditorConfig and #14182 instead.

**Comment by pchote** [comment]
Updated.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (reaperrr) [inline .github/PULL_REQUEST_TEMPLATE.md]: This seems to be missing an 'ask' after the 'can' now.

**Review by reaperrr — DISMISSED** [review]

**Comment by reaperrr** [comment]
Looks good to me except that one issue pointed out in the diff.

**Comment by pchote** [comment]
Done.

**Review by reaperrr — APPROVED** [review]

**Review by penev92 — APPROVED** [review]

**Pull request merged**
